### PR TITLE
Add helm chart for deploying GMC itself

### PR DESCRIPTION
**Pull request by lianhao**
## Description

Add helm chart for deploying GMC itself.

## Issues

issue #248 .

## Type of change

List the type of change like below. Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds new functionality)

## Dependencies

n/a.

## Tests

Describe the tests that you ran to verify your changes.

### CODE REVIEW

**Comment by zhlsunshine** [comment]
How to make all environment variables to use the same format, such as `${KIND_MOUNT_DIR}`?

File: .github/workflows/gmc-helm.yaml [hunk 1]
```diff
@@ -56,14 +56,30 @@ jobs:
       - name: Install GMC
         run: |
           echo "should_cleanup=true" >> $GITHUB_ENV
-          ## IMPORTANT NOTICE: need to replace gmc images with {{env.DOCKER_REGISTRY}} and {{env.VERSION}}
-          cd microservices-connector
-          if ! helm install --create-namespace --namespace {{ env.SYSTEM_NAMESPACE}} --wait \
+          USER_ID=$(whoami)
+          HFTOKEN=$(cat /home/$USER_ID/.cache/huggingface/token)
+          pushd microservices-connector/helm
```
Comment (irisdingbj) [inline .github/workflows/gmc-helm.yaml:61]: Can we add this info into microservices-connector/helm/README.md as well?

**Review by irisdingbj — COMMENTED** [review]
a minor comment, otherwise LGTM, Thanks.

File: .github/workflows/gmc-helm.yaml [hunk 2]
```diff
@@ -56,14 +56,30 @@ jobs:
...
+          # replace the pull policy "IfNotPresent" with "Always" for gmc-router
+          sed -i "s|imagePullPolicy:.*|imagePullPolicy: Always|" gmc-router.yaml
+          # replace image for GenAI component manifests
+          find manifests_common/ -type f -name "*.yaml" -exec sed -i "s|image: \"opea|image: \"${DOCKER_REGISTRY}opea|" {} \;
+          # replace the pull policy "IfNotPresent" with "Always" for GenAI component manifests
+          find manifests_common/ -type f -name "*.yaml" -exec sed -i "s|imagePullPolicy: IfNotPresent|imagePullPolicy: Always|" {} \;
+          # replace the mount dir "path: /mnt/model" with "path: $CHART_MOUNT"
+          find manifests_common/ -type f -name "*.yaml" -exec sed -i "s|path: /mnt/opea-models|path: $KIND_MOUNT_DIR|" {} \;
+          # replace huggingface token
+          find manifests_common/ -type f -name '*.yaml' -exec sed -i "s#insert-your-huggingface-token-here#${HFTOKEN}#g" {} \;
```
Comment (KfreeZ) [inline .github/workflows/gmc-helm.yaml:73]: also consider google API key?

File: microservices-connector/helm/README.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+To use this GMC helm chart, you need to have a Kubernetes cluster and helm installed. If you don't have helm(version >= 3.15) installed, you can follow the instructions [here](https://helm.sh/docs/intro/install/).
+
+This helm chart will install the following components of GMC:
+
+- GMC CRD
+- GenAI Components and GMC Router manifests
+- GMC Manager
+
+**NOTE: Because helm doesn't support updating/deleting CRD, you need to manually delete the CRD before upgrading the GMC helm chart.**
+
```
[the remark this answers is not in the document]
Comment (lianhao) [reply microservices-connector/helm/README.md:16]: will fix along with HF token

File: .github/workflows/gmc-helm.yaml (hunk 2 above)
Comment (lianhao) [reply .github/workflows/gmc-helm.yaml:73]: google_api_key and google_cse_id is exposed as secret. To use that, we need to use a different set of event 'pull_request_target' and need quite a lots of change, also using . I think we should fix this along with go e2e CI in another PR, need @daisy-ycguo 's input here

File: .github/workflows/gmc-helm.yaml (hunk 1 above)
Comment (lianhao) [reply .github/workflows/gmc-helm.yaml:65]: done

File: microservices-connector/helm/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (lianhao) [reply microservices-connector/helm/README.md:16]: done

File: .github/workflows/gmc-helm.yaml [hunk 4]
```diff
@@ -8,13 +8,13 @@ on:
     branches: [main]
     types: [opened, reopened, ready_for_review, synchronize] # added `ready_for_review` since draft is skipped
     paths:
-      - manifests/common/**
```
Comment (daisy-ycguo) [inline .github/workflows/gmc-helm.yaml:11]: why deleting manifests/common/** from the trigger paths?

File: .github/workflows/gmc-helm.yaml (hunk 2 above)
Comment (daisy-ycguo) [reply .github/workflows/gmc-helm.yaml:77]: Yes. Better to use two PRs. Using secrets is more complex for CI workflows.

File: .github/workflows/gmc-helm.yaml (hunk 4 above)
Comment (lianhao) [reply .github/workflows/gmc-helm.yaml:11]: I don't think helm deployment methods will be different if there is any changes in manifest/common. We requires user to copy the manifest/common directory before doing 'helm install' now.

**Review by zhlsunshine — APPROVED** [review]
LGTM

File: .github/workflows/gmc-helm.yaml (hunk 4 above)
Comment (daisy-ycguo) [reply .github/workflows/gmc-helm.yaml:11]: Will manifest/common changes cause failure of GMC helm deployment ?

**Review by daisy-ycguo — APPROVED** [review]
lgtm

File: .github/workflows/gmc-helm.yaml (hunk 4 above)
Comment (lianhao) [reply .github/workflows/gmc-helm.yaml:11]: nope, it won't affect this helm chart, which is designed to install GMC/manifests itself, no matter what common manifest files is, the helm chart will only treat them as a bunch of raw files and install them as into the configmap.

However, there could be errors when you launch a GMC pipeline using those manifest files. But that could be catch by `GMC go e2e test`

**Pull request merged**